modules/cassandra/src/test/java/org/testcontainers/containers/CassandraContainerTest.java

```diff
@@ -0,0 +1,132 @@
+package org.testcontainers.containers;
+
+import com.datastax.driver.core.Cluster;
+import com.datastax.driver.core.ResultSet;
+import com.datastax.driver.core.Row;
+import com.datastax.driver.core.Session;
+import lombok.extern.slf4j.Slf4j;
+import org.junit.Test;
+import org.testcontainers.containers.output.Slf4jLogConsumer;
+import org.testcontainers.containers.wait.CassandraQueryWaitStrategy;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+
+/**
+ * @author Eugeny Karpov
+ */
+@Slf4j
+public class CassandraContainerTest {
+
+    private static final String TEST_CLUSTER_NAME_IN_CONF = "Test Cluster Integration Test";
+
+    @Test
+    public void testSimple() {
+        try (CassandraContainer cassandraContainer = (CassandraContainer) new CassandraContainer()
+                .withLogConsumer(new Slf4jLogConsumer(log))) {
+            cassandraContainer.start();
+            ResultSet resultSet = performQuery(cassandraContainer, "SELECT release_version FROM system.local");
+            assertTrue("Query was not applied", resultSet.wasApplied());
+            assertNotNull("Result set has no release_version", resultSet.one().getString(0));
+        }
+    }
+
+    @Test
+    public void testSpecificVersion() {
+        String cassandraVersion = "3.0.15";
+        try (CassandraContainer cassandraContainer = (CassandraContainer) new CassandraContainer("cassandra:" + cassandraVersion)
+                .withLogConsumer(new Slf4jLogConsumer(log))) {
+            cassandraContainer.start();
+            ResultSet resultSet = performQuery(cassandraContainer, "SELECT release_version FROM system.local");
+            assertTrue("Query was not applied", resultSet.wasApplied());
+            assertEquals("Cassandra has wrong version", cassandraVersion, resultSet.one().getString(0));
+        }
+    }
+
+    @Test
+    public void testConfigurationOverride() {
+        try (CassandraContainer cassandraContainer = (CassandraContainer) new CassandraContainer()
+                .withConfigurationOverride("cassandra-test-configuration-example")
+                .withLogConsumer(new Slf4jLogConsumer(log))) {
+            cassandraContainer.start();
+            ResultSet resultSet = performQuery(cassandraContainer, "SELECT cluster_name FROM system.local");
+            assertTrue("Query was not applied", resultSet.wasApplied());
+            assertEquals("Cassandra configuration is not overridden", TEST_CLUSTER_NAME_IN_CONF, resultSet.one().getString(0));
+        }
+    }
+
+    @Test(expected = ContainerLaunchException.class)
+    public void testEmptyConfigurationOverride() {
+        try (CassandraContainer cassandraContainer = (CassandraContainer) new CassandraContainer()
```

Comment:
Did you try to incorperate the proposed fix for omitting explicit cast mentioned by @bsideup?

Reply:
Yes, I did. Pushed commit several minutes ago. So no excess casts anymore :+1: